### ISSUE TITLE
Exception ArgumentOutOfRange when using ToStereo extension on a mono SampleSource

### ISSUE THREAD

**Issue by Zarto** [filoe/cscore#160]
As soon as I try to read samples from SampleSource an ArgumentOutOfRange is throwed. 
It only happens when I use the ToStereo extension on a mono file.

Am I doing a mistake or is it a CSCore bug ?

Here's a snippet of code that throws the exception : 
    
```
 private ISampleSource sampleSource;

        public Form1()
        {
            InitializeComponent();

            sampleSource = CodecFactory.Instance.GetCodec(@"C:\Users\Utilisateur\Desktop\Mono\mono.mp3").ToSampleSource().ToStereo().ChangeSampleRate(44100);

            // These examples work !
            // sampleSource = CodecFactory.Instance.GetCodec(@"C:\Users\Utilisateur\Desktop\Mono\mono.mp3").ToSampleSource().ChangeSampleRate(44100);
            // sampleSource = CodecFactory.Instance.GetCodec(@"C:\Users\Utilisateur\Desktop\Jingles\stereo.mp3").ToSampleSource().ToStereo().ChangeSampleRate(44100);


            const int bufferSize = 1024;

            int read = 0;
            float[] buffer = new float[bufferSize];

            do
            {
                read = sampleSource.Read(buffer, 0, bufferSize);
                if (read > 0) Console.WriteLine(buffer[0] + "/");
            } while (read > 0);
        }
```

The StackTrace : 
```
   à CSCore.SampleAggregatorBase.Read(Single[] buffer, Int32 offset, Int32 count)
   à CSCore.Streams.MonoToStereoSource.Read(Single[] buffer, Int32 offset, Int32 count)
   à CSCore.Streams.SampleConverter.SampleToIeeeFloat32.Read(Byte[] buffer, Int32 offset, Int32 count)
   à CSCore.DSP.DmoResampler.Read(Byte[] buffer, Int32 offset, Int32 count)
   à CSCore.Streams.SampleConverter.IeeeFloatToSample.Read(Single[] buffer, Int32 offset, Int32 count)
   à Test_Mono_CSCore.Form1..ctor() dans C:\Users\Utilisateur\Documents\Visual Studio 2015\Projects\Test Mono CSCore\Test Mono CSCore\Form1.cs:ligne 37
   à Test_Mono_CSCore.Program.Main() dans C:\Users\Utilisateur\Documents\Visual Studio 2015\Projects\Test Mono CSCore\Test Mono CSCore\Program.cs:ligne 19
   à System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   à System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   à Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   à System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   à System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   à System.Threading.ThreadHelper.ThreadStart()
```

**Comment by filoe** [comment]
Whats the exact message of the exception?

**Comment by Zarto** [comment]
The message is in french. I try to translate this for you : The exception System.ArgumentOutOfRangeException hasn't been handled.
“Specified argument was out of the range of valid values”

Oh and thank you for this great library !

Here's the original message : 

```
L'exception System.ArgumentOutOfRangeException n'a pas été gérée
  HResult=-2146233086
  Message=L'argument spécifié n'était pas dans les limites de la plage des valeurs valides.
Nom du paramètre : count
  ParamName=count
  Source=CSCore
  StackTrace:
       à CSCore.SampleAggregatorBase.Read(Single[] buffer, Int32 offset, Int32 count)
       à CSCore.Streams.MonoToStereoSource.Read(Single[] buffer, Int32 offset, Int32 count)
       à CSCore.Streams.SampleConverter.SampleToIeeeFloat32.Read(Byte[] buffer, Int32 offset, Int32 count)
       à CSCore.DSP.DmoResampler.Read(Byte[] buffer, Int32 offset, Int32 count)
       à CSCore.Streams.SampleConverter.IeeeFloatToSample.Read(Single[] buffer, Int32 offset, Int32 count)
       à Test_Mono_CSCore.Form1..ctor() dans C:\Users\Utilisateur\Documents\Visual Studio 2015\Projects\Test Mono CSCore\Test Mono CSCore\Form1.cs:ligne 37
       à Test_Mono_CSCore.Program.Main() dans C:\Users\Utilisateur\Documents\Visual Studio 2015\Projects\Test Mono CSCore\Test Mono CSCore\Program.cs:ligne 19
       à System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       à System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       à Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       à System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       à System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       à System.Threading.ThreadHelper.ThreadStart()
  InnerException: 

```

**Comment by Zarto** [comment]
I forgot to mention that it works if I initialize and play the SoundOut with sampleSource.ToWaveSource()
Only the read function fails ...

**Comment by filoe** [comment]
Thanks, I'll have a look at it.

**Comment by filoe** [comment]
I've had a look at it. Does this bug occur with the current master branch?
As far as I could see, this bug fixed with 2a393882.
